CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+cmake_minimum_required(VERSION 3.21)
```

Comment:
Would it make more sense to put this in the `ggml` subdirectory? Given it only builds GGML targets (currently at least)?

Reply:
Possibly, at least a large part however it seems premature. I want this to be something that can be invoked from the root directory

Reply:
I'm ok with that, we should sort out the `Makefile` issue below though since it will appear to be meant for building the project.

Are there steps/instructions on building the different backends using this?